library/ssl_tls.c

```diff
@@ -5744,13 +5744,24 @@ int mbedtls_ssl_parse_certificate( mbedtls_ssl_context *ssl )
         /*
          * Main check: verify certificate
          */
-        ret = mbedtls_x509_crt_verify_restartable(
+
+        if( authmode == MBEDTLS_SSL_VERIFY_EXTERNAL )
+        {
+            if( NULL != ssl->conf->f_vrfy )
```

Comment:
Two points:
1. We might want to fail if `ssl->conf->f_vrfy` is unset, instead of silently succeeding.
2. The semantics of `f_vrfy` in the new authmode needs to be documented. The existing callbacks are expected to return `0` even if they find an issue with the CRT, and to instead the verification flags accordingly. In other words, all but non-fatal failures must be communicated through verification flags, but not through the return value. This is different with the suggested re-purposing of `f_vrfy` here, which - at least with the current code - would need to return a non-zero value for the verification to fail.